internal/resolution/entities/bundle_entity.go

```diff
@@ -121,14 +119,35 @@ func (b *BundleEntity) ChannelName() (string, error) {
 	if err := b.loadChannelProperties(); err != nil {
 		return "", err
 	}
-	return b.channelProperties.ChannelName, nil
+	return b.channel.ChannelName, nil
 }
 
-func (b *BundleEntity) ChannelProperties() (*ChannelProperties, error) {
+func (b *BundleEntity) Channel() (*property.Channel, error) {
 	if err := b.loadChannelProperties(); err != nil {
 		return nil, err
 	}
-	return b.channelProperties, nil
+	return b.channel, nil
+}
+
+func (b *BundleEntity) Replaces() (string, error) {
+	if err := b.loadReplaces(); err != nil {
+		return "", err
+	}
+	return b.replaces.Replaces, nil
+}
+
+func (b *BundleEntity) loadReplaces() error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	if b.replaces == nil {
+		// TODO: move property name to constant
```

Comment:
should we address this todo?

Reply:
Removing this TODO as it is tied to: https://github.com/operator-framework/operator-controller/pull/275#discussion_r1265236837